docs/guides/building/arm-cross-compile.md

```diff
@@ -0,0 +1,162 @@
+4. Install [Tauri using Cargo]: `cargo install tauri-cli`.
+5. Install necessary tools for your frontend framework, like npm and Next.js.
+
+:::info OpenSSL Workaround for WSL on Windows
+
+If using WSL on Windows and encountering OpenSSL-related errors, add the following line to `Cargo.toml` in `<project-root>/src-tauri/Cargo.toml` within the `[dependencies]` section:
+```toml
+openssl-sys = {version = "0.9.66", features = ["vendored"]}
+```
+:::
+
+### Cross-Compiling
+
+Now, let's cross-compile the Tauri application for ARM:
+
+1. Install Rust target for ARM: `rustup target add armv7-unknown-linux-gnueabihf`.
+2. Install the linker for ARM version: `sudo apt install gcc-arm-linux-gnueabihf`.
+3. Open or create the file `<project-root>/.cargo/config.toml` and add the following:
+   ```toml
+   [target.armv7-unknown-linux-gnueabihf]
+   linker = "arm-linux-gnueabihf-gcc"
+   ```
+4. Enable armhf in the package manager: `sudo dpkg --add-architecture armhf`.
+
+:::info Adjusting Package Sources
+
+On Debian, this step isn't necessary, but on other distributions, you might need to edit `/etc/apt/sources.list` to include the ARM architecture variant. For example, add these lines:
+```bash
+deb [arch=armhf] http://ports.ubuntu.com/ubuntu-ports jammy main restricted
+deb [arch=armhf] http://ports.ubuntu.com/ubuntu-ports jammy-updates main restricted
+deb [arch=armhf] http://ports.ubuntu.com/ubuntu-ports jammy universe
+deb [arch=armhf] http://ports.ubuntu.com/ubuntu-ports jammy-updates universe
+deb [arch=armhf] http://ports.ubuntu.com/ubuntu-ports jammy multiverse
+deb [arch=armhf] http://ports.ubuntu.com/ubuntu-ports jammy-updates multiverse
+deb [arch=armhf] http://ports.ubuntu.com/ubuntu-ports jammy-backports main restricted universe multiverse
+deb [arch=armhf] http://ports.ubuntu.com/ubuntu-ports jammy-security main restricted
+deb [arch=armhf] http://ports.ubuntu.com/ubuntu-ports jammy-security universe
+deb [arch=armhf] http://ports.ubuntu.com/ubuntu-ports jammy-security multiverse
+```
+After making changes, verify if the armhf architecture is still enabled in the package manager by re-running the command from Step 4.
+
+:::
+
+5. Update the package info: `sudo apt-get update && sudo apt-get upgrade -y`.
+6. Install webkitgtk for ARM: `sudo apt install libwebkit2gtk-4.0-dev:armhf`.
+7. Set the PKG_CONFIG_SYSROOT_DIR for the appropriate arch: `export PKG_CONFIG_SYSROOT_DIR=/usr/arm-linux-gnueabihf/`.
+8. Build the app by specifying the desired ARM version: `cargo tauri build --target armv7-unknown-linux-gnueabihf`.
+
+### Final Notes
+
+- The static website can be found in the `out` folder.
+- Rust may try to run your app after compiling it, which will fail on non-ARM platforms. Ignore this error, as the .deb package will be correctly built.
+
+## Automatic Cross-Compilation using a GitHub Action
+
+For automated ARM executable builds on GitHub, we'll use the [arm-runner-action] created by [Paul Guyot].
+
+### Setup
```

Comment:
I can't look into this section too deeply today but 2 things users might end up having problems with are that it doesn't use tauri-action, understandibly so since it's incompatible with the arm action, and secondly that it uploads it as an artifact and not to the release what we do in the other guides.

Because of that i'd almost prefer mirroring the local approach 🤔 not sure though.
Also maybeee we should pull this, and the actions from the cross-platform and macos guides into a new GitHub Action build page but we can do that in a separate PR or just in the new docs.

Reply:
It would be great if you could use the tauri action instead. It would be great, especially because you would have:
1. Everything in one action
2. Automatic Release draft creation, which is a feature I really enjoy.

I need to admit that I was not able to set up this action alone, the creator of the Action, Paul, helped me out a lot and fixed most of the issues I encountered. You are right, this might fit better somewhere else. Also, something I forgot to mention is that the Action takes about 3 hours to create an executable on Raspirus. My project is not too big, but also not very efficient, so maybe it's a bit less for others. Still, it's a very long time and I usually let it run before going to sleep.

Reply:
Maybe I should remove the action entirely. It's a bit weird, as you said